database/migrations/2014_10_13_100000_create_email_addresses_table.php

```diff
@@ -17,5 +20,21 @@ public function up()
             $table->dateTime('undeliverable_at')->nullable(); // If email is undeliverable, need to change to update
             $table->timestamps();
         });
+
+        // Direct SQL to avoid execution of custom attribute getter/setters
+        $sql = <<<SQL
+insert into email_addresses (`email`, `user_id`, `primary`, `verified_at`, `created_at`, `updated_at`)
+select email, id, 1, email_verified_at, current_timestamp, current_timestamp from users
+SQL;
+
+        DB::insert($sql);
+
+        Schema::table('users', function (Blueprint $table) {
+            $table->dropColumn('email');
+        });
+
+        Schema::table('users', function (Blueprint $table) {
+            $table->dropColumn('email_verified_at');
+        });
```

Comment:
This migrates any existing emails to the `email_addresses` table and then removes the corresponding source columns from the `users` table.  I would much rather something break loudly due to an expected DB field missing than having code not fail, but not work properly because its using the wrong fields.

Reply:
That's a great implemention. I was wondering if there was going to be a way to do that and didn't consider doing it in the migration. Thank you!